direnvrc

```diff
@@ -197,8 +197,19 @@ _nix_argsum_suffix() {
 }
 
 nix_direnv_watch_file() {
+  # shellcheck disable=2016
+  log_error '`nix_direnv_watch_file` is deprecated - use `watch_file`'
   watch_file "$@"
-  nix_watches+=("$@")
+}
+
+_nix_direnv_watches() {
+  # DIRENV_WATCHES is json | gzip (without header and trailer) | b64url
+  # gzip header: https://datatracker.ietf.org/doc/html/rfc1952#page-6
+  tr -- "-_" "+/" <<< "$DIRENV_WATCHES" \
+    | base64 --decode \
+    | (echo -ne "\x1f\x8b\x08\x00\x00\x00\x00\x00"; cat /dev/stdin) \
```

Comment:
It's cool to re-use direnv's watches here, I am only a bit on the fence with depending on jq. Nix is often introduced in new teams through direnv and those folks might not have all dependencies installed, which can cause friction. I wonder if we can make somehow a poly fill that uses nix for json parsing as a fallback?

```suggestion
    | (echo -ne "\x1f\x8b\x08\x00\x00\x00\x00\x00"; base64 --decode) \
```

Reply:
Also do we have base64 on macOS? We might have to depend on openssl base64?

Reply:
pushed with polyfill, macOS has base64